### ISSUE TITLE
Release pipeline should upload to packages.grpc.io

### ISSUE THREAD

**Issue by mehrdada** [grpc/grpc#18307]
Currently, only daily builds are published to packages.grpc.io. It'd be great to unify the release process such that we also archive official release points on that website, which will automatically resolve issues like https://github.com/grpc/grpc/issues/18260. In fact, it would simplify the life of the release manager if they could click a button and the release showed up there and then language owners uploaded from there, instead of hunting for the build package files. This should be quite straightforward and probably requires only minimal changes to `grpc_publish_packages.sh` script.

**Comment by peter-edge** [comment]
Just a note: once this is done, it would probably be helpful to update the installation instructions in the markdown file in this repository to point out that the binaries can be downloaded from here (along with an example using curl and tar). Right now the instructions just say to build from source. Thanks!

**Comment by nicolasnoble** [comment]
You're welcome to come back help us with this, @mehrdada :D

**Comment by mehrdada** [comment]
@nicolasnoble it's really hard to test things without proper access :)

**Comment by peter-edge** [comment]
Let me know if there's anything I can do to help!

**Comment by jtattermusch** [comment]
I'm generally not opposed to this, but we're recently added some steps (e.g. nuget package signing) that would need to be figured out first.

Did you mean this script?
https://github.com/grpc/grpc/blob/master/tools/internal_ci/linux/grpc_publish_packages.sh

**Comment by mehrdada** [comment]
@jtattermusch yes, I think it is mostly parameterized. Somehow we need to make Kokoro pass the right branch name.

**Comment by pedgeio** [comment]
Any updates?

**Comment by gnossen** [comment]
No updates as of yet. But we're planning work for the next quarter soon, so it's possible it'll be prioritized.

**Comment by jtattermusch** [comment]
What we currently have is automatic uploading of nightly builds to our jfrog artifactory nuget channel, which is something that could relatively easily be done for other languages too.
See https://github.com/grpc/grpc/tree/master/src/csharp#nuget-development-feed-nightly-builds

The challenge with pushing the official packages is that those packages need to be signed (needs t be done on a different VM because of infrastructure limitations) for some languages (C#) and during the build you'd need to specify a "flag" that this is THE build that should upload the official packages (we don't want official stable packages accidentally uploaded by mistake and also for some languages, you can only upload given package version once, so there's no way to fix an accidentally pushed wrong package)

**Comment by gnossen** [comment]
Not stale.

**Comment by jtattermusch** [comment]
We acknowledge this to be a valid feature request but since no one has come forward to implement this, we are going to close it for now while maintaining the label/Help wanted to indicate that this can be picked up as future work.
